services/Donation.js

```diff
@@ -171,6 +171,52 @@ class DonationService {
       throw new ErrorHandler(e, 500);
     }
   }
+
+  /**
+   * @description Records a bulk donation not done via Flutterwave
+   * @param {object} data
+   */
+  static async recordBulkDonation(payload) {
+    const { name, email, phone_number, ...referralDetails } = payload;
+    try {
+      // This should mirror Flutterwave's webhook payload
+      const transactionPayload = {
+        tx_ref: referralDetails.tx_ref,
+        flw_ref: "Not applicable",
+        amount: referralDetails.amount,
+        currency: "NGN",
+        charged_amount: referralDetails.amount,
+        app_fee: 0,
+        merchant_fee: 0,
+        processor_response: "Successful",
+        auth_model: "INTERNET_BANKING",
+        status: "successful",
+        payment_type: "direct",
+        account_id: 1111111 // An arbitrary value just to mirror Flutterwave
+      };
+
+      const donorPayload = {
+        id: 1111111, // Arbitrary value to mirror FLW
+        name,
+        email,
+        phone_number
+      };
+
+      const donationData = {
+        transactionId: Math.floor(Math.random() * 9000000) + 1000000, // Arbitrary value,
+        donor: donorPayload,
+        event: "charge.completed", // As would be returned by FLW
+        transaction: transactionPayload
+      };
+
+      const newDonation = new Donation(donationData);
+      await newDonation.save();
+      return newDonation;
+    } catch (err) {
+      Logger.error(`${err.message} when creating bulk transaction of ${name}`);
```

Comment:
Please update the log message to be `... when creating manual transaction...` instead of bulk.

Reply:
Sorry I somehow missed this.